### PR TITLE
Add an exception for COPYRIGHT and NOTICE files in the same way as LICENSE

### PR DESCRIPTION
**Pull request by vinnybod**
Addresses https://github.com/bazelbuild/rules_jvm_external/issues/1106

### CODE REVIEW

**Comment by shs96c** [comment]
Thank you for the PR. It looks great, and I'll happily merge it, but could you please sign the Google CLA? If you [follow the link in the Google CLA bot's comment](https://github.com/bazelbuild/rules_jvm_external/pull/1107#issuecomment-2070440784) you'll be able to do that.

**Comment by vinnybod** [comment]
Thanks @shs96c , I'm just waiting on the authorized signer's signature. Hopefully will go through today.

**Comment by vinnybod** [comment]
CLA is signed, but its still not going through. May still be waiting for it to be processed on Google's side

**Comment by shs96c** [comment]
@jin any idea why the CLA check isn't passing?

@vinnybod, the CLA check is complaining that `<v***e​@confluent.io>` hasn't signed the CLA. Did you use that email address? Maybe it's something to do with that?

**Comment by vinnybod** [comment]
Yes, I got a corporate CLA signed and submitted and I am in the Google group.
According to https://opensource.google/documentation/reference/cla#cla-accepted , it might take a couple days for it to process.

**Comment by shs96c** [comment]
Great. I'll re-run the check tomorrow and see if it passes. Thank you for being patient!

**Comment by vinnybod** [comment]
@shs96c I'm fine merging this since it solves my immediate issue.

But I do want to continue this discussion a bit more, because I can see myself needed to add more files to this list in the near future.

Why should file `NOTICE` in `app-export` be excluded just because it also exists in `lib-export`? I wonder if this is actually a bug.
It seems like if I want to include `NOTICE` in `app-export`, that should happen regardless of whether `lib-export` contains it or not. Does that make sense?

**Comment by vinnybod** [comment]
Actually, I think we should wait to merge this. I am seeing that in some cases this does not do what I need.
For example I have a java_export where I Add a `NOTICE`, but instead of my `NOTICE`, it is including the `NOTICE`, from my scala dependency.

**Comment by shs96c** [comment]
The `LICENSE` files are kept because it's often a legal requirement to keep them. The default `ZipOutputStream` in java forbids multiple files with the same name, but in theory what we need to do is write each license to the zip and preserve it. For Buck, I wrote [a custom zip implementation that did this](https://github.com/facebook/buck/blob/main/src/com/facebook/buck/util/zip/AppendingZipOutputStreamImpl.java)

**Comment by vinnybod** [comment]
Okay, so I now believe this to be fine to merge. The issue I was running into was due to a misconfiguration with `excluded_workspaces` that was causing some extra files in the jars from dependents that should not have been there.

TL;DR: This solves the immediate issue of files being excluded from the `-project` jar when one of its deps contains a file of the same name. Treating `COPYRIGHT` and `NOTICE` files the same as we do `LICENSE`.

**Review by shs96c — APPROVED** [review]
LGTM. Thank you!
